pramen/api/src/main/scala/za/co/absa/pramen/api/Source.scala

```diff
@@ -57,6 +66,28 @@ trait Source extends ExternalChannel {
     */
   def getData(query: Query, infoDateBegin: LocalDate, infoDateEnd: LocalDate, columns: Seq[String]): SourceResult
 
+  /**
+    * Returns the incremental data between specified offsets. The offset intervals could be half open,
+    * e.g. only offsetFrom or offsetFrom is specified.
+    *
+    * If an information date is provided and available at the source, the query will be limited to that date.
+    *
+    * <ul>
+    * <li> When both `offsetFrom` from and `offsetTo` are passed the source should return offsets using an inclusive interval
+    *   (offsetFrom <= offset <= offsetTo) </li>
+    * <li> When only `offsetFrom` is present the source should return offsets using an exclusive interval interval
+    *   (offset > offsetFrom)</li>
```

Comment:
When we want to query new data the caller would specify only `offsetFrom`, and the query is going to look like:
```sql
SELECT * FROM table WHERE offset > offsetFrom
```
(exclusive)

When we want to do a rerun for a day, we get the minimum and maximum offsets for the day and run:
```sql
SELECT * FROM table WHERE offset >= offsetFrom offset <= offsetTo
```
(inclusive)

The last case, when only `offsetTo` is available might not be used in practice. Added it for completion. Potentially it can be used to query the old database for all data that was already loaded. Maybe for reconciliation or something like this. In this case only inclusive interval makes sense for such queries.
```sql
SELECT * FROM table WHERE offset <= offsetTo
```
(inclusive)

I've could have added a flag, to make the caller choose, or split this method in 2, just for top 2 cases. But then it would require the implementer of a source to implement 2 very similar methods. 

Added this reasoning to the method documentation